### PR TITLE
CYA - Client stricter config

### PR DESCRIPTION
**Pull request by amitdahan**
### 🔦 Summary
Minor `tsconfig.json` tweaks for the client (TS) template.
**Note:** Making this template stricter uncovered the lack of `@types/puppeteer` in generated client projects.
This also adds `@types/puppeteer` to `jest-yoshi-preset` since it plants `page: Puppeteer.Page` in the global. 
The addition of type declarations for `yoshi-template-intro` is temporary until we can get https://github.com/wix/yoshi/pull/1487 merged.

### 🗺️ Test Plan
Template passed ✅

### CODE REVIEW

**Comment by amitdahan** [comment]
I also wasn't so sure about `noEmit`, wdyt?

**Comment by ronami** [comment]
> I also wasn't so sure about `noEmit`, wdyt?

I like it, if a developer checks for type errors by running `npx tsc` directly they don't need to remember passing `--noEmit`

File: packages/create-yoshi-app/templates/client/typescript/src/external-types.d.ts [hunk 1]
```diff
@@ -2,3 +2,5 @@ interface Window {
   __LOCALE__: string;
   __BASEURL__: string;
 }
+
+declare module 'yoshi-template-intro';
```
Comment (ranyitz) [inline packages/create-yoshi-app/templates/client/typescript/src/external-types.d.ts]: `yoshi-template-intro` is something that our users should remove right after they are generating the project. Now they also need to remove it from here.

Maybe we can think of another solution that wouldn't add another file to the templates?

Comment (amitdahan) [reply packages/create-yoshi-app/templates/client/typescript/src/external-types.d.ts]: Yes, I forgot to mention, this is temporary (`noImplicitAny` forced this) until we can get https://github.com/wix/yoshi/pull/1487 merged and types will be shipped correctly so it's not needed. Updating the PR accordingly.

**Comment by amitdahan** [comment]
Thanks! 🙏
Glad these tests were here to catch this!

**Comment by amitdahan** [comment]
Thanks! Will definitely be on alert 😀

**Comment by amitdahan** [comment]
Closes https://github.com/wix/yoshi/issues/1427 :)

**Pull request merged**